Mathlib/Algebra/Order/Ring/Basic.lean

```diff
@@ -61,23 +61,31 @@ theorem pow_add_pow_le (hx : 0 ≤ x) (hy : 0 ≤ y) (hn : n ≠ 0) : x ^ n + y
         exact mul_le_mul_of_nonneg_left (ih (Nat.succ_ne_zero k)) h2
 
 @[bound]
-theorem pow_le_one : ∀ n : ℕ, 0 ≤ a → a ≤ 1 → a ^ n ≤ 1
+theorem pow_le_one₀ : ∀ {n : ℕ}, 0 ≤ a → a ≤ 1 → a ^ n ≤ 1
```

Comment:
This PR also changes the implicitness of some lemmas. In particular it makes `n` implicit in lemmas which do not have a hypothesis relating to `n`. Please note this change in the PR description, together with a justification for it.

Reply:
Done

Reply:
Can you explain what you mean by "where the explicit arguments are located is basically random throughout the library"? As mentioned above, locally the location of the explicit argument is pretty clear and consistent, so I don't follow the argument here.

Reply:
I'm not questioning swapping the order - that part is fine - it's the change in implicitness that I'm concerned about. The other lemmas in this PR take `n` as an implicit argument exactly when it is mentioned in one of the other hypotheses, showing consistency. As such, the PR description should justify why the implicitness has been changed.

Reply:
Mm okay, that's convincing.